### PR TITLE
Support two clusters of bidirectional replication.

### PR DESCRIPTION
**Pull request by freemindLi**
### What problem does this PR solve? 
Support two clusters of bidirectional replication.  
A <-> B

### What is changed and how it works?
- surport ddl one-way synchronization
- surport dml Mutual synchronization

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by GregoryIan** [comment]
Cool！Great Work, thanks @freemindLi

**Comment by GregoryIan** [comment]
@july2993 @WangXiangUSTC  PTAL

File: pkg/loader/load.go [hunk 1]
```diff
@@ -595,6 +668,11 @@ func (b *batchManager) execAccumulatedDMLs() (err error) {
 }
 
 func (b *batchManager) execDDL(txn *Txn) error {
+
+	if !b.ddlSync {
```
Comment (july2993) [inline pkg/loader/load.go]: can we handle sync or not sync outside loader in `draienr/syncer.go` directly?

File: pkg/loader/executor.go [hunk 2]
```diff
@@ -50,6 +57,20 @@ func (e *executor) withBatchSize(batchSize int) *executor {
...
+func newSyncInfo(ddlSync bool, markStatus bool, channelId int64) *syncInfo {
+	sync := &syncInfo{
+		markStatus: markStatus,
+		ddlSync:    ddlSync,
+		channelId:  channelId,
+	}
+	return sync
+}
+
+func (e *executor) setsyncInfo(sync *syncInfo) *executor {
```
Comment (WangXiangUSTC) [inline pkg/loader/executor.go]: seems that don't need to return executor

File: drainer/config.go [hunk 3]
```diff
@@ -129,6 +132,9 @@ func NewConfig() *Config {
 	fs.Int64Var(&cfg.InitialCommitTS, "initial-commit-ts", -1, "if drainer donesn't have checkpoint, use initial commitTS to initial checkpoint, will get a latest timestamp from pd if setting to be -1")
 	fs.StringVar(&cfg.Compressor, "compressor", "", "use the specified compressor to compress payload between pump and drainer, only 'gzip' is supported now (default \"\", ie. compression disabled.)")
 	fs.IntVar(&cfg.SyncerCfg.TxnBatch, "txn-batch", 20, "number of binlog events in a transaction batch")
+	fs.BoolVar(&cfg.SyncerCfg.MarkStatus, "mark-status", false, "set mark or not ")
+	fs.BoolVar(&cfg.SyncerCfg.DdlSync, "ddl-sync", false, "sync ddl or not")
+	fs.Int64Var(&cfg.SyncerCfg.ChannelId, "channel-id", 0, "sync channel id ")
```
Comment (WangXiangUSTC) [inline drainer/config.go]: how about adding more details about these three config

File: pkg/loader/model.go [hunk 4]
```diff
@@ -184,6 +184,23 @@ func (dml *DML) updateSQL() (sql string, args []interface{}) {
 	return
 }
 
+func updateMarkSQL(columns []string, Values map[string]interface{}) (string, []interface{}) {
+
+	//sql := fmt.Sprintf("REPLACE INTO %s(%s) VALUES(%s)",MarkTableName, buildColumnList(columns),holderString(len(columns)))
```
Comment (WangXiangUSTC) [inline pkg/loader/model.go]: can delete this line

File: drainer/config.go [hunk 5]
```diff
@@ -65,6 +65,9 @@ type SyncerConfig struct {
 	IgnoreSchemas     string             `toml:"ignore-schemas" json:"ignore-schemas"`
 	IgnoreTables      []filter.TableName `toml:"ignore-table" json:"ignore-table"`
 	TxnBatch          int                `toml:"txn-batch" json:"txn-batch"`
+	MarkStatus        bool               `toml:"mark-status" json:"mark-status"`
+	DdlSync           bool               `toml:"ddl-sync" json:"ddl-sync"`
```
Comment (WangXiangUSTC) [inline drainer/config.go]: how about use https://github.com/pingcap/tidb-tools/blob/06b27bff7c2e29aac542b2e7197a52ae96221142/pkg/binlog-filter/filter.go#L85 to support more function? it is not necessary, we can do this later

**Comment by july2993** [comment]
@freemindLi #867 will allow when the column number of downstream table mismatch with current schema. you can help take a look too.

**Pull request closed without merging**